### PR TITLE
[HttpKernel] Denormalize request data using the csv format when using "#[MapQueryString]" or "#[MapRequestPayload]" (except for content data)

### LINKED ISSUE #59104: [MapQueryString] Weird behavior while mapping integer variable
### Symfony version(s) affected

7.1.9

### Description

While trying to map a GET request to a DTO which contains a nullable integer variable, symfony does not initialize the entity and also does not respond to the request with the validation error.

### Create the following DTO ### 
```
use Symfony\Component\Validator\Constraints as Assert;

readonly class FilterDTO {
  public function __construct(
    #[Assert\NotBlank(allowNull: true, message: "The ID must not be blank.")]
    #[Assert\Positive(message: "The ID must be positive.")]
    public ?int $id = null,

    #[Assert\NotBlank(allowNull: true, message: "The name must not me blank.")]
    #[Assert\NoSuspiciousCharacters]
    public ?string $name = null,
  )
  {

  }
}
```
### Create the following controller ### 
```
#[Route(path: "/test", name: "test", methods: ["GET"], format: "json")]
public function readAll(
  #[MapQueryString] FilterDTO $filter = new FilterDTO
): JsonResponse 
{
   dd($filter);
   return $this->json(data: [$filter]);
}
```
###  In the browser, postman or any other tool, make the requests to the endpoint /test ###
The first request will generate the following output (/test):
ok_result
The second request will generate the following output (/test?id=x):
wrong_result

### Possible Solution

_No response_

### Additional Context

If using any other constraint, like Length, in a string field, symfony correctly validates the field and returns a 404 error, but with integer fields it does not respond with the validation error and also does not create the DTO (the fields become not initialized).

### PR DESCRIPTION
**Pull request by ovidiuenache**
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | #59104 #50904
| License       | MIT

This fixes the scenario mentioned above where using `#[MapQueryString]` or `#[MapRequestPayload]` (except request content) with at least one readonly property with a type mismatch error leads to getting an object with uninitialized properties. The only properties that are set are the ones that are assigned a value via a request parameter and are NOT readonly. Moreover, if the corresponding query parameter is not provided for non-readonly property A and there is at least one other readonly property B that has a type mismatch then A will still be returned as uninitialized (even if it has a default value).

Shortly put, the instantiation fails and the values of the properties cannot be set later on.

Examples
```
class FilterDto {
    public function __construct(
        public readonly ?int $id = null,
        public readonly ?string $name = null,
        public ?string $description = null,
    ) {
    }
}

GET https://127.0.0.1:8000/test?id=x&name=test
id: ? ?int
name: ? ?string
description: ? ?string

GET https://127.0.0.1:8000/test?id=x&name=test&description=desc
id: ? ?int
name: ? ?string
description: "desc"
```

The complete list of steps to reproduce this is provided in #59104.

The reason why this happens is because we are disabling the type enforcement of the denormalizer in the `Symfony\Component\HttpKernel\Controller\ArgumentResolver\RequestPayloadValueResolver` class and when we eventually end up in the `validateAndDenormalize` method of the `Symfony\Component\Serializer\Normalizer\AbstractObjectNormalizer` class we ignore the type mismatch because of:
```
if ($context[self::DISABLE_TYPE_ENFORCEMENT] ?? $this->defaultContext[self::DISABLE_TYPE_ENFORCEMENT] ?? false) {
    return $data;
}
``` 

Thus, we get a type error when trying to create the object and we fall back to
`$reflectionClass->newInstanceWithoutConstructor();`

Then, based on the provided request data, we attempt to set the values of the properties but this process fails for the readonly properties so they stay uninitialized.

As discussed with @nicolas-grekas during [the hackathon at SymfonyCon Vienna 2024](https://live.symfony.com/2024-vienna-con/) the solution here is to stop disabling the type enforcement of the denormalizer. However, this alone is not enough because then we won't be able to use anything but string since this is the type that comes in the request so we also need to set the denormalization format to either `csv` or `xml`.

This comment from `the Symfony\Component\Serializer\Normalizer\AbstractObjectNormalizer` sheds some light on why:
```
// In XML and CSV all basic datatypes are represented as strings, it is e.g. not possible to determine,
// if a value is meant to be a string, float, int or a boolean value from the serialized representation.
// That's why we have to transform the values, if one of these non-string basic datatypes is expected.
```

We avoid `xml` due to some special formatting that occurs so the proposed solution uses `csv`.

Basically, we start using type enforcement + csv format where non-string values are transformed.

### CODE REVIEW

**Comment by ovidiuenache** [comment]
The coding standards check is failing due to existing violations in the files I've updated. It says [here](https://symfony.com/doc/current/contributing/code/pull_requests.html) that I should `Never fix coding standards in some existing code as it makes the code review more difficult;` so I left it as is and will fix the existing flagged issues on request.

File: src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php [hunk 1]
```diff
@@ -161,7 +160,7 @@ private function mapQueryString(Request $request, string $type, MapQueryString $
             return null;
         }
 
-        return $this->serializer->denormalize($data, $type, null, $attribute->serializationContext + self::CONTEXT_DENORMALIZE);
+        return $this->serializer->denormalize($data, $type, CsvEncoder::FORMAT, $attribute->serializationContext + self::CONTEXT_DENORMALIZE);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php]: Maybe we can inline the value to save loading an extra class?

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php (hunk 1 above)
Comment (ovidiuenache) [reply src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php]: 5d1014d9d56e9c43a71e13c1d65122edcdfa5438

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @ovidiuenache.
